pkg/kubelet/prober/prober.go

```diff
@@ -132,16 +134,13 @@ func (pb *prober) runProbeWithRetries(ctx context.Context, probeType probeType,
 	}
 	return result, output, err
 }
-
-func (pb *prober) runProbe(ctx context.Context, probeType probeType, p *v1.Probe, pod *v1.Pod, status v1.PodStatus, container v1.Container, containerID kubecontainer.ContainerID) (probe.Result, string, error) {
-	timeout := time.Duration(p.TimeoutSeconds) * time.Second
-	if p.Exec != nil {
-		klog.V(4).InfoS("Exec-Probe runProbe", "pod", klog.KObj(pod), "containerName", container.Name, "execCommand", p.Exec.Command)
-		command := kubecontainer.ExpandContainerCommandOnlyStatic(p.Exec.Command, container.Env)
-		return pb.exec.Probe(pb.newExecInContainer(ctx, container, containerID, command, timeout))
-	}
-	if p.HTTPGet != nil {
-		req, err := httpprobe.NewRequestForHTTPGetAction(p.HTTPGet, &container, status.PodIP, "probe")
+func (pb *prober) runHTTPProbe(p *v1.Probe, status v1.PodStatus, container *v1.Container, timeout time.Duration) (probe.Result, string, error) {
+	var err error
+	var result probe.Result
+	var output string
+	var req *http.Request
+	for _, podIP := range status.PodIPs {
```

Comment:
this can be a breaking change, since an app that binds to only the primary IP will start failing after this change

Reply:
No, the primary family (the one in status.PodIPs[0]) is tried first, and if it doesn't result in "connection refused" (i.e. a success, warning or any other error), the second is not tested. This is tested with my test program.

Reply:
if the primary fails and the second works it succeeds?
it is a change in behavior if it does it

Reply:
Yes

Reply:
there are very subtle things that can break, as I said https://github.com/kubernetes/kubernetes/pull/123112#issuecomment-1931077451 , I didn't see any report complaining about this, we should try to avoid premature optimization if this implies noticeable changes on behaviors on users